spec/unit/environment/with_modules_spec.rb

```diff
@@ -0,0 +1,88 @@
+require 'spec_helper'
+require 'r10k/environment'
+
+describe R10K::Environment::WithModules do
+  subject do
+    described_class.new(
+      'release42',
+      '/some/nonexistent/environmentdir',
+      'prefix_release42',
+      {
+        :type             => 'bare',
+        :modules          => {
+          'puppetlabs-stdlib' => '6.0.0',
+          'puppetlabs-concat' => '6.1.0',
+          'puppetlabs-exec'   => '0.5.0',
+        }
+      }.merge(subject_params)
+    )
+  end
+
+  # Default no additional params
+  let(:subject_params) { {} }
+
+  describe "dealing with Puppetfile module conflicts" do
+    context "with no module conflicts" do
+      it "validates when there are no conflicts" do
+        mod = double('module', :name => 'nonconflict')
+        expect(subject.puppetfile).to receive(:load)
+        expect(subject.puppetfile).to receive(:modules).and_return [mod]
+        expect { subject.validate_no_module_conflicts }.not_to raise_error
+      end
+    end
+
+    context "with module conflicts and default behavior" do
+      it "does not raise an error" do
+        mod = double('duplicate-stdlib', :name => 'stdlib')
```

Comment:
Is there some magic that strips off the `puppetlabs-` part of the name? Otherwise, I'm not sure how this would be a duplicate of the environment version.

Reply:
Yeah, there's some magic. `name` != what the user put in the Puppetfile as the title.
https://github.com/puppetlabs/r10k/blob/master/lib/r10k/module/base.rb#L51

Reply:
👍🏼  thanks